### PR TITLE
Update pre-commit, remove Python 3.4 from tests

### PR DESCRIPTION
**Pull request by sjaensch**
- make sure pre-commit hooks are installed whenever people run tests
- also ran fix-encoding-pragma and reorder-python-imports on all files

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ docs:
 install:
 	pip install .
 
-test:
+test: install-hooks
```
Comment (macisamuele) [inline Makefile]: Is this needed? ``tox`` will install all the needed dependencies.
Maybe you like to run ``pre-commit`` checks during tests (ie. ``pre-commit run flake8 --all-files``)

Comment (sjaensch) [inline Makefile]: flake8 already runs as part of `make test`. The problem with running pre-commit during make test is the virtualenv setup; I'm sure there's a solution, I just didn't want to change that as well in this PR.
install is a separate target and has nothing to do with install-hooks. What's your concern here?

Comment (macisamuele) [inline Makefile]: The concern was related to adding a new dependency on the make target that actually is not need; but according to your explanation it seems needed! So just drop what I said 😄

Comment (sjaensch) [inline Makefile]: Yeah the way it works for bravado-core is that it installs pre-commit in a separate virtualenv. Moving it to install it during the normal test run (in the same virtualenv) and then running flake8 causes ist to install and run three times (once for each Python version we test).

Comment (macisamuele) [inline Makefile]: Thanks for the more detailed explaination.
🚢

**Pull request merged**